Tests/Shared.Specs/ExecutionTimeAssertionsSpecs.cs

```diff
@@ -94,6 +94,30 @@ public void When_the_execution_time_of_an_action_is_less_or_equal_to_a_limit_it_
             //-----------------------------------------------------------------------------------------------------------
             act.Should().NotThrow();
         }
+
+        [Fact]
+        public void When_action_runs_indefinitely_it_should_be_stopped_and_throw_if_there_is_less_or_equal_condition()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            Action someAction = () =>
+            {
+                // lets cause a deadlock
+                var semaphore = new Semaphore(0, 1); // my weapon of choice is a semaphore
+                semaphore.WaitOne(); // oops
+            };
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action act = () => someAction.ExecutionTime().Should().BeLessThan(100.Milliseconds());
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            act.Should().Throw<XunitException>();
```

Comment:
Shouldn't you at least verify part of the message. Because if your code doesn't work, you'll still get an XUnit exception if I'm not mistaken. Same for the other tests.

Reply:
well, if my code doesnt work, the test never finishes to run, but fair enough, better to be safe